### ISSUE TITLE
Unable to parse audit logs if set custom name_format

### ISSUE THREAD

**Issue by manuasir** [wazuh/wazuh#3342]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
|3.9.0 | Logcollector | Manager | Packages | Linux |

Hello guys,

There is a problem when having a custom `name_format` value in the `/etc/audit/auditd.conf`. It provokes the audit log messages to have a field `node=VALUE`, which seems that is not currently supported. I'm attaching here a few audit logs:
```
node=hids.socib.co type=USER_START msg=audit(1558101002.000:17948): pid=16484 uid=0 auid=0 ses=1797 msg='op=PAM:session_open grantors=pam_loginuid,pam_keyinit,pam_limits,pam_systemd acct="root" exe="/usr/sbin/crond" hostname=? addr=? terminal=cron res=success'
node=hids.socib.co type=CRED_REFR msg=audit(1558101002.001:17949): pid=16484 uid=0 auid=0 ses=1797 msg='op=PAM:setcred grantors=pam_env,pam_unix acct="root" exe="/usr/sbin/crond" hostname=? addr=? terminal=cron res=success'
node=hids.socib.co type=CRED_DISP msg=audit(1558101002.019:17950): pid=16484 uid=0 auid=0 ses=1797 msg='op=PAM:setcred grantors=pam_env,pam_unix acct="root" exe="/usr/sbin/crond" hostname=? addr=? terminal=cron res=success'
```

and some logs after setting debug mode:

```
2019/05/17 18:10:45 ossec-logcollector[12101] read_fullcommand.c:28 at read_full command(): DEBUG: Running full command 'netstat -tulpn | sed 's/([[:alnum:]]+\ )\ +[[:digit:]]+\ +[[:digit:]]+\ +(.):([[:digit:]])\ +([0-9.:*]\ +).+\ ([[:digit:]]/[[:alnum:]-])./\1 \2 == \3 == \4 \5/' | sort -k 4 -g | sed 's/ == (.) ==/:\1/' | sed 1,2d'
2019/05/17 18:10:45 ossec-logcollector[12101] read_fullcommand.c:28 at read_full command(): DEBUG: Running full command 'last -n 20'
2019/05/17 18:10:45 ossec-logcollector[12101] read_audit.c:93 at read_audit(): E RROR: Discarding audit message because of invalid syntax.
2019/05/17 18:10:45 ossec-logcollector[12101] read_audit.c:120 at read_audit(): DEBUG: Read 1 lines from /var/log/audit/audit.log
```

Regards

**Comment by crolopez** [comment]
The error occurs because Logcollector waits for the line to start with `type=... msg=audit(...`.

Related line:

https://github.com/wazuh/wazuh/blob/993b7826f7df762f9262807065ab3c482c7903dd/src/logcollector/read_audit.c#L102

**Comment by mkraft12** [comment]
Any updates on if/when this will be fixed?

**Comment by selivan** [comment]
I've got the same problem with wazuh-agent and wazuh-manager version 3.9.3-1.
Had to disable this part of config to get rid of this error messages:

```xml
<localfile>
    <log_format>audit</log_format>
    <location>/var/log/audit/audit.log</location>
  </localfile>
```
So seems auditd integration doesn't work.

**Comment by pjmorr** [comment]
I have the same issue and have even tried changing the `/etc/audit/auditd.conf` and commented out the name_format line as well as leaving it in with value of none instead of the default `hostname` value.

Even after host reboots I am still receiving the same error in wazuh /var/ossec/logs/ossec.log.

I can say that we seem to only see this error on amazonlinux nodes that are kubernetes hosts (not sure if this makes a difference).
I can log into other systems in the environment and they are reporting correctly.

Any ideas?

**Comment by doyelese** [comment]
any update on this? @JcabreraC @vikman90 ?

**Comment by vikman90** [comment]
Hi all,

Sorry for such a delay. The `audit` log reader requires that the log starts with `type=`. 

I've just opened PR #12676 to fix this problem. In the meantime, as a workaround, you can use the `syslog` format. The difference is that this reader won't join logs with the same audit ID into the same event.

```xml
<localfile>
  <log_format>syslog</log_format>
  <location>/var/log/audit/audit.log</location>
</localfile>
```

Sorry for the inconvenience.
Best regards.

**Comment by vikman90** [comment]
We've just merged PR #12676 into branch 4.3 for the next version 4.3.0.

Best.
